plugin/LaserDevice.h

```diff
@@ -35,18 +35,46 @@ namespace smode
   {
   public:
     LaserDevice(const DeviceIdentifier& identifier)
-      : ControlDevice(identifier), dacPointsPerSecond(10000), latency(166), targetFps(60), distancePerPoint(0.1), blankDelayPoints(10), radiansPerPoint(0.6),
+      : ControlDevice(identifier), dacPointsPerSecond(10000), latencyPoints(166), targetFps(60), blankDelayPoints(10), distancePerPoint(0.1), anglePerPoint(0.6),
       callback_data(std::make_shared<CallbackData>())
     {
       dacPointsPerSecond.setParent(this);
-      latency.setParent(this);
+      latencyPoints.setParent(this);
       targetFps.setParent(this);
       distancePerPoint.setParent(this);
       blankDelayPoints.setParent(this);
-      radiansPerPoint.setParent(this);
+      anglePerPoint.setParent(this);
     }
     LaserDevice() {}
 
+    // Object
+    bool hasVariableConstraints() const override
+      {return true;}
+
+    void variableChangedCallback(Object* variable, Object* changedObject) override
+    {
+      if (isInitialized() && isRenderingServiceCurrent()) {
```

Comment:
@francis-maes we noticed that the `variableChangedCallback` is called prior to `initializeDevice`. My guess is that the reason for this is so that the initialization process can use the previously saved values for each parameter. Is this correct? I just wanted to clarify as this is what we are currently expecting within our implementation.

Reply:
variableChangedCallback can be called at anytime. In this case, there is a non trivial mechanism to store the values of the variables inside the configuration. When the device is created, the variables are first settled from the config and then initializeDevice() is being called. It is thus a good practice to test the initialized state. 
Note: since the variable `initialized `is protected, you can simplify as:
`if (initialized && isRenderingServiceCurrent())`

Reply:
Great, thanks for clarifying!